src/CodeGen_LLVM.h

```diff
@@ -465,6 +471,26 @@ class CodeGen_LLVM : public IRVisitor {
      * an arbitrary number of vectors.*/
     virtual llvm::Value *interleave_vectors(const std::vector<llvm::Value *> &);
 
+    /** Description of an intrinsic function overload. Overloads are resolved
```

Comment:
The data structure one likely wants here is an unordered_map indexed by both the name and the types. The key could be a string perhaps even a fixed length type given that we can likely put an upper bound on the number of arguments to an overloaded intrinsic. The goal here is to make sure compile time efficiency is preserved.

Reply:
I want to consider allowing type promotion though, which means the key lookup isn't just a straightforward check for equality. There's already a bit of this going on which would make this tricky even now.

Also, these vectors are pretty small, I wouldn't be surprised if the benefit of them being in a vector outweighs lookup efficiency.